addons/pt/saft/extensions.go

```diff
@@ -959,6 +968,109 @@ var extensions = []*cbc.Definition{
 			},
 		},
 	},
+	{
+		Key: ExtKeySourceBilling,
+		Name: i18n.String{
+			i18n.EN: "Document Source",
+			i18n.PT: "Origem do documento",
+		},
+		Desc: i18n.String{
+			i18n.EN: here.Doc(`
+				SAF-T's ~SourceBilling~ (Origem do documento) field specifies the source of the document in the
+				accounting system. GOBL provides the ~pt-saft-source-billing~ extension to set this value in
+				your documents. By default, GOBL will set this to "P" (produced in the application).
+
+				The table below shows the supported source billing codes:
+
+				| Code | Name                                    | Description                                    |
+				| ---- | --------------------------------------- | ---------------------------------------------- |
+				| ~P~  | Document produced in the application     | Document generated within the current system   |
+				| ~I~  | Integrated document from another app     | Document integrated from another application   |
+				| ~M~  | Document from recovery or issued manually | Document recovered from backup or issued manually |
+
```

Comment:
Sorry, but I'm having real difficulty understanding the naming and the descriptions. "Source Billing" isn't clear to me what is implied and I can't get my head around "Produced" or "Integrated". In the table, the name and the description basically say the same thing, so not much value there.

What I'd suggest for the name is just putting something simple in, like "Produced", and the providing a description for what that actually means and implies.

I appreciate the source XML document uses the `SourceBilling` name though, and that these codes are just weird, but I wonder if there is an easier name. Even just reversing might help so that we end the key with a noun: `billing-source`

Reply:
Addressed in [6f6f0e3](https://github.com/invopop/gobl/pull/582/commits/6f6f0e3e2259d2ffce80f21bc2f43d7729a1e2d4). I finally decide to use just `source` for the extension, for brevity. LMK!